### ISSUE TITLE
renv::install default type arg in renv 1.0.8

### ISSUE THREAD

**Issue by pascal-sauer** [rstudio/renv#1992]
We have a custom CRAN-like R package server where we host most of our in-house packages, just as source packages, no binaries. Since switching to renv 1.0.8 (with 1.0.7 it was fine) we cannot install these source packages like before. I noticed that renv::install's default arg `type = NULL`  should according to documentation be the same as `type = getOption('pkgType')`, but it makes a difference:
```
> options(repos = c(pik = "https://rse.pik-potsdam.de/r/packages", CRAN = "https://cran.rstudio.com"))

> renv::install('mrmagpie') 
Error: package 'mrmagpie' is not available
Traceback (most recent calls last):
12: renv::install("mrmagpie")
11: records <- renv_retrieve_impl(packages) at install.R#210
10: for (package in packages)
      handler(package, renv_retrieve_impl_one(package)) at retrieve.R#120
 9: for (package in packages)
      handler(package, renv_retrieve_impl_one(package)) at retrieve.R#120
 8: record <- withCallingHandlers(
      renv_available_packages_latest(package),
      error = function(err) stopf("package '%s' is not available", package)
    ) at retrieve.R#200
 7: record <- withCallingHandlers(
      renv_available_packages_latest(package),
      error = function(err) stopf("package '%s' is not available", package)
    ) at retrieve.R#200
 6: stopf("package '%s' is not available", package) at available-packages.R#416
 5: stop(sprintf(fmt, ...), call. = call.) at utils-format.R#3
 4: .handleSimpleError(function (err) 
    {
        ...
    }, "package 'mrmagpie' is not available", base::quote(NULL))
 3: h(simpleError(msg, call))
 2: record <- withCallingHandlers(
      renv_available_packages_latest(package),
      error = function(err) stopf("package '%s' is not available", package)
    ) at retrieve.R#200
 1: stop(sprintf(fmt, ...), call. = call.) at utils-format.R#3

> renv::install('mrmagpie', type = getOption('pkgType'))
The following package(s) will be installed:
- mrmagpie [1.52.1]
These packages will be installed into ...

> getOption('pkgType')
[1] "source"
```
Am I missing something, or is that a bug? Thanks for your help! :+1: 

This is R 4.3.2 on a linux machine.

**Comment by kevinushey** [comment]
Thanks for the bug report! I believe this should be fixed in renv 1.0.9, which was just submitted as a hotfix to CRAN. Can you let me know if the issue appears to be resolved in your case?

**Comment by pascal-sauer** [comment]
The issue is fixed with renv 1.0.9, thanks a lot!
